# HG changeset patch
# User Joel Keay <keay@ou.edu>
# Date 1258142376 18000
#      Fri Nov 13 14:59:36 2009 -0500
# Node ID 29f4f8c9b3f1335a64893e3373afcd7379ec74cb
# Parent  2b29f3472e2012ca0c1b6ff92a583c80d43f572b
allow single, int16 and uint16 image data

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-11-13  Joel Keay  <keay@ou.edu>
+
+	* graphics.h.in (image::properties::init): Also allow single,
+	int16, and uint16 data types.
+
 2009-11-13  Jaroslav Hajek  <highegg@gmail.com>
 
 	* ov-base.h (unary_mapper_t): Move to octave_base_value.
 	* ov.h: Partially revert to c657c056240d.
 	(FORWARD_MAPPER): Update to use octave_base_value::map.
 	(octave_value::map): Ditto.
 	* mappers.cc: Revert to c657c056240d.
 	* data.cc: Ditto.
diff --git a/src/graphics.h.in b/src/graphics.h.in
--- a/src/graphics.h.in
+++ b/src/graphics.h.in
@@ -3183,18 +3183,21 @@ public:
     END_PROPERTIES
 
   protected:
     void init (void)
       {
 	xdata.add_constraint (2);
 	ydata.add_constraint (2);
 	cdata.add_constraint ("double");
+	cdata.add_constraint ("single");
 	cdata.add_constraint ("logical");
 	cdata.add_constraint ("uint8");
+	cdata.add_constraint ("uint16");
+	cdata.add_constraint ("int16");
 	cdata.add_constraint (dim_vector (-1, -1));
 	cdata.add_constraint (dim_vector (-1, -1, 3));
       }
 
   private:
     void update_xdata (void)
     { 
       octave_idx_type iw = (get_cdata ().dims ())(1) - 1;
